core/templates/components/button-directives/create-activity-button.component.ts

```diff
@@ -82,14 +82,15 @@ export class CreateActivityButtonComponent implements OnInit {
    showUploadExplorationModal(): void {
      this.explorationCreationService.showUploadExplorationModal();
    }
-   ngOnInit(): void {
+
+   async ngOnInit(): Promise<void> {
```

Comment:
This function does not seem like it returns anything, right?

Reply:
Also, more generally, avoid making changes to source files for frontend test coverage PRs unless it is absolutely necessary.

Reply:
Yep, it does not.

I added the changes as, `canCreateCollections` was not being set to `true` before the control reached [this](https://github.com/oppia/oppia/pull/12936/files#diff-18b57c9d65ff111b9d10b87671ac5aadd027ffeaf361d8e341ad4342c7f05f98L100) line hence inside the ngOnInit function, the value of `canCreateCollections` was false and we were not able to enter [this](https://github.com/oppia/oppia/pull/12936/files#diff-18b57c9d65ff111b9d10b87671ac5aadd027ffeaf361d8e341ad4342c7f05f98L100) else block.

Is there any other way to test this code, without using `await`?

Reply:
However, I think you might have uncovered a possible bug in the code.

getUserInfoAsync() only sets `canCreateCollections` after the backend call completes. But we seem to use `canCreateCollections` outside the success callback so it could execute even if the call does not complete.

To answer your question, yep, it is possible. You'll just need to move more of the logic in to the success callback of the async function.

```ts
    this.userService.getUserInfoAsync().then(userInfo => {
      this.canCreateCollections = userInfo.canCreateCollections();
      this.userIsLoggedIn = userInfo.isLoggedIn();
      // If the user clicked on a 'create' button to get to the dashboard,
      // open the create modal immediately (or redirect to the exploration
      // editor if the create modal does not need to be shown).
      if (this.urlService.getUrlParams().mode === 'create') {
        if (!this.canCreateCollections) {
          this.explorationCreationService.createNewExploration();
        } else {
          this.initCreationProcess();
        }
      }
     });
```

That said, I'm fine with the current approach you have. It does the same thing and looks better. :)

Reply:
async ngOnInit is not a good practice (this is perfectly fine code, but its counter intuitive). Please undo this change. Angular calls this function before the view renders,  and it won't wait for the promise to resolve. Using async await will make error handling a pain here. Also, this will cause the life cycle hooks to end in different order.

Here is some more context: https://github.com/angular/angular/issues/17420